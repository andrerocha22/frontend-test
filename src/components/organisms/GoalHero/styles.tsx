import styled from 'styled-components';

export const Hero = styled.div`
  display: flex;
  flex-direction: column;
  width: 100%;
  align-content: flex-start;
  justify-content: center;
`;
