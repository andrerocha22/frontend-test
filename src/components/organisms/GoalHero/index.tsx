import React from 'react';
import { Icon } from '../../atoms/icon';
import { PrimaryText, SecondaryText } from '../../atoms/typography';

import { Hero } from './styles';

interface GoalHeroProps {
  title: string;
  description: string;
  iconSrc: string;
}

const GoalHero: React.FC<GoalHeroProps> = ({ title, description, iconSrc }) => {
  return (
    <Hero data-test="component-goal-hero">
      <Icon src={iconSrc} />
      <PrimaryText>{title}</PrimaryText>
      <SecondaryText>{description}</SecondaryText>
    </Hero>
  );
};

export default GoalHero;
