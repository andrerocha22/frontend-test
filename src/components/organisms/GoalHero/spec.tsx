import React from 'react';
import { shallow } from 'enzyme';
import { findByTestAttr } from '../../../test/testUtils';
import 'jest-styled-components';

import GoalHero from './index';

const defaultProps = {
  title: 'test-title',
  description: 'test-description',
  iconSrc: 'icon-src',
};

const setup = (props = {}) => {
  const setupProps = { ...defaultProps, ...props };
  return shallow(<GoalHero {...setupProps} />).dive();
};

describe('test goal hero', () => {
  test('renders without error', () => {
    const wrapper = setup();
    const component = findByTestAttr(wrapper, 'component-goal-hero');
    expect(component.length).toBe(1);
  });

  test('goal hero snapshot test ', () => {
    const wrapper = setup();
    const component = findByTestAttr(wrapper, 'component-goal-hero');
    expect(component).toMatchSnapshot();
  });
});
