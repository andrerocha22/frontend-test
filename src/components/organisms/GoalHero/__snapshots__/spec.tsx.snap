// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test goal hero goal hero snapshot test  1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
  -webkit-align-content: flex-start;
  -ms-flex-line-pack: start;
  align-content: flex-start;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

<div
  className="c0"
  data-test="component-goal-hero"
>
  <Component
    src="icon-src"
  />
  <styled.p>
    test-title
  </styled.p>
  <styled.p>
    test-description
  </styled.p>
</div>
`;
